### DISCUSSION TITLE
DirectStorage metadata generator only generate constants and enums

### DISCUSSION THREAD

**Discussion by mlarouche** [microsoft/win32metadata#926] [Q&A] [answered]
Hello!

I am trying to generate the metadata for DirectStorage in order to consume it in [Zig](https://ziglang.org/) with [zigwin32](https://github.com/marlersoft/zigwin32).

But so far, I was only able to generate the enums and constants contained in the header files. I have extract the DirectStorage nupkg package manually in a `DirectStorage` folder. 

I have posted my .proj, `global.json` and `main.cpp` in this gist: 
https://gist.github.com/mlarouche/405bd3696e06e22758c7babaeba5d0cc

Any help to get the full metadata generated would be really helpful.

Thanks :)

**Comment by mlarouche** [comment]
Should I file an issue instead?

**Comment by riverar** [comment, marked as answer]
@mlarouche I made some minor tweaks to your project to make it standalone, but otherwise it spits out a valid winmd upon inspection. Are there APIs missing? [DirectStorage is nano-COM API](https://github.com/microsoft/DirectStorage/blob/main/Docs/DeveloperGuidance.md#using-directstorage), so you'll generally be interacting with things like DStorageGetFactory, IDStorageFactory, and IDStorageFile which are all present.

```xml
<?xml version="1.0" encoding="utf-8"?>
<Project Sdk="Microsoft.Windows.WinmdGenerator">
  <PropertyGroup>
    <OutputWinmd>bin\Microsoft.Direct3D.DirectStorage.winmd</OutputWinmd>
    <WinmdVersion>1.0.0.0</WinmdVersion>
    <GenerateAssemblyVersionInfo>false</GenerateAssemblyVersionInfo>
    <AdditionalIncludes>DirectStorage\Include\DirectStorage</AdditionalIncludes>
    <ShowEmitWinmdOutputDetails>true</ShowEmitWinmdOutputDetails>
  </PropertyGroup>

  <ItemGroup>
    <Headers Include="$(PkgMicrosoft_Direct3D_DirectStorage)\native\include\*.h" />
    <ImportLibs Include="$(PkgMicrosoft_Direct3D_DirectStorage)\native\lib\x64\dstorage.lib" />
    <Partition Include="main.cpp">
      <TraverseFiles>@(Headers)</TraverseFiles>
      <Namespace>Microsoft.Direct3D.DirectStorage</Namespace>
      <ExcludeFromCrossarch>true</ExcludeFromCrossarch>
    </Partition>

    <PackageReference Include="Microsoft.Direct3D.DirectStorage" Version="1.0.2" GeneratePathProperty="true">
      <ExcludeAssets>All</ExcludeAssets>
    </PackageReference>
  </ItemGroup>
</Project>
```

**Comment by mlarouche** [comment]
That did the trick thanks a lot :)
